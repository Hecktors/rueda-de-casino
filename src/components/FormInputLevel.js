import { useEffect, useState } from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ArrowRightIcon, ArrowDownIcon } from '../components/Icons'

FormInputLevel.propTypes = {
  levelName: PropTypes.string.isRequired,
  levelMoves: PropTypes.array.isRequired,
  selectedMoveIDs: PropTypes.array.isRequired,
  updateUserInput: PropTypes.func.isRequired,
}

export default function FormInputLevel({
  levelName,
  levelMoves,
  selectedMoveIDs,
  updateUserInput,
}) {
  const [isOpen, setIsOpen] = useState()
  const hasInputMove = levelMoves.some((move) => {
    return selectedMoveIDs.includes(move.id)
  })

  useEffect(() => {
    !hasInputMove && isOpen && setIsOpen(false)
  }, [hasInputMove]) // eslint-disable-line react-hooks/exhaustive-deps

  const listItems = levelMoves.map((move) => (
    <li key={move.id}>
      <label className={selectedMoveIDs.includes(move.id) ? 'isChecked' : ''}>
        <input
          name="move"
          value={move.id}
          onChange={updateUserInput}
          type="checkbox"
          checked={selectedMoveIDs.includes(move.id)}
        />{' '}
        {move.name}
      </label>
    </li>
  ))

  function toogleFormInputLevel() {
    setIsOpen(!isOpen)
  }

  return (
    <FormInputLevelStyled isOpen={isOpen} isActive={hasInputMove}>
      <h3 onClick={toogleFormInputLevel}>
        {isOpen ? <ArrowDownIcon /> : <ArrowRightIcon />}{' '}
        <span className="level-name">{levelName}</span>
      </h3>
      <ul>{listItems}</ul>
    </FormInputLevelStyled>
  )
}

const FormInputLevelStyled = styled.div`
  background-color: var(--color-bg-accordion);
  border-radius: 5px;

  input {
    display: none;
  }

  h3 {
    cursor: pointer;
    font-size: 1rem;
    padding: 7px;
    display: flex;
    justify-content: space-between;

    & .level-name {
      color: ${({ isActive }) =>
        isActive ? 'var(--color-primary)' : 'var(--color-secondary)'};
      flex-grow: 1;
      text-align: center;
    }

    svg {
      transform: scale(1.5);
      fill: ${({ color }) => color};
      fill: var(--color-secondary);
    }
  }

  ul {
    line-height: 1.7;
    padding: 0 9px;
    overflow: hidden;
    height: ${(props) => (props.isOpen ? 'auto' : 0)};
  }

  li {
    &:first-of-type {
      padding-top: 1px;
    }
    &:last-of-type {
      padding-bottom: 10px;
    }

    label {
      font-size: 0.875rem;
      color: var(--color-secondary);
      display: inline-block;
      width: 100%;

      &:hover {
        cursor: pointer;
      }
    }
  }

  .isChecked {
    color: var(--color-primary);
  }
`
