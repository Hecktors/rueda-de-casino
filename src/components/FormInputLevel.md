```jsx
<FormInputLevel
  levelName="level"
  levelMoves={[
    {
      id: 'm1',
      name: 'la prima',
      steps: 2,
      filename: 'la_prima.mp3',
      videoId: 'SFxy0cWwEtg',
      videoStart: '20',
    },
    {
      id: 'm3',
      name: 'sombrero con una mano',
      steps: 2,
      filename: 'sombrero_con_una_mano.mp3',
      videoId: 'fAMaQ_3HEUE',
      videoStart: '1',
    },
  ]}
  moveIDs={[]}
  updateUserInput={() => {}}
/>
<FormInputLevel
  levelName="level"
  levelMoves={[
    {
      id: 'm1',
      name: 'la prima',
      steps: 2,
      filename: 'la_prima.mp3',
      videoId: 'SFxy0cWwEtg',
      videoStart: '20',
    },
    {
      id: 'm3',
      name: 'sombrero con una mano',
      steps: 2,
      filename: 'sombrero_con_una_mano.mp3',
      videoId: 'fAMaQ_3HEUE',
      videoStart: '1',
    },
  ]}
  moveIDs={['m1', 'm3']}
  updateUserInput={() => {}}
/>
```
