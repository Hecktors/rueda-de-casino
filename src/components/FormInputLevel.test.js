import ReactDOM from 'react-dom'
import FormInputLevel from './FormInputLevel'
import { render } from '@testing-library/react'
const moves = [
  {
    id: 1,
    name: 'la prima',
    steps: 2,
    filename: 'la_prima.mp3',
  },
  {
    id: 2,
    name: 'sombrero',
    steps: 2,
    filename: 'sombrero.mp3',
  },
]

describe('FormInputLevel', () => {
  it('renders Accordion', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <FormInputLevel
        name="test accordion"
        moves={moves}
        levelName="level"
        selectedMoves={[]}
        userInput={[2]}
        updateUserInput={() => {}}
      />,
      div
    )
    expect(div.firstChild).toMatchSnapshot()
  })
})
