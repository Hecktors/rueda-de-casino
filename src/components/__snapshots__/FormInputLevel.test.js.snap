// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputLevel renders Accordion 1`] = `
<div
  class="FormInputLevel__FormInputLevelStyled-fylxmi-0 XIaHt"
>
  <h3>
    <svg>
      arrow_right.svg
    </svg>
     
    <span
      class="level-name"
    >
      level
    </span>
  </h3>
  <ul>
    <li>
      <label
        class=""
      >
        <input
          name="move"
          type="checkbox"
          value="1"
        />
         
        la prima
      </label>
    </li>
    <li>
      <label
        class=""
      >
        <input
          name="move"
          type="checkbox"
          value="2"
        />
         
        sombrero
      </label>
    </li>
  </ul>
</div>
`;
